docs/docs/grpc.md

```diff
@@ -18,12 +18,134 @@ main = runGRpcApp 8080 "helloworld" quickstartServer
 
 ## Building a client
 
--- TODO:
+Right now there are two options for building clients: using records or with `XTypeApplications`. To give a proper introduction to both options let's consider in detail an example client for the following services:
+
+```protobuf
+service Service {
+  rpc getPerson (PersonRequest) returns (Person);
+  rpc newPerson (Person) returns (PersonRequest);
+  rpc allPeople (google.protobuf.Empty) returns (stream Person);
+}
+```
+
+Regardless of the approach we decide to use, we can construct a basic CLI for the client this way:
+
+```haskell
+import System.Environment
+
+main :: IO ()
+main = do
+  let config = grpcClientConfigSimple "127.0.0.1" 8080 False
+  Right client <- setupGrpcClient' config
+  args <- getArgs
+  case args of
+    ["watch"]       -> watching client
+    ["get", idp]    -> get client idp
+    ["add", nm, ag] -> add client nm ag
+    _               -> putStrLn "unknown command"
+```
+
+Where `watch`, `get` and `add` are the only valid 3 commands that our CLI is going to accept and call each respective service.
 
 ### Using records
 
--- TODO:
+This option is a bit more verbose but it's also more explicit with the types and _"less magic"_ than the one with `TypeApplications`.
+
+We need to define a new record type (hence the name) that declares the services our client is going to consume:
```

Comment:
It's important to emphasize that the names of the fields **must** coincide with the methods in the service.